### PR TITLE
[195-sixgill] update alert status

### PR DESCRIPTION
**Pull request by tamarsix**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Added comment for updating alert status.
Small bug fixes.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X ] No

### CODE REVIEW

**Comment by tamarsix** [comment]
Docker image need to be updated after this will be merged: https://github.com/demisto/dockerfiles/pull/7031

**Comment by kgal-pan** [comment]
Hey @tamarsix - See below comments:

1) I see that you've modified the `commonfields.id` of a CybersixgillActionableAlerts.yml from `Cybersixgill_Actionable_Alerts` to `Cybersixgill_Actionable_Alerts_dev` and name of the command from/to the same. Modifying the `commonfields.id` will introduce breaking changes and is not recommended. What is the reason this change is necessary?

2) I see that a new command was added (`cybersixgill-update-alert-status`). This change requires release notes and a bump in the version of the Pack. Please run the following command to bump the minor version and fill out the newly-created file in `./ReleaseNotes/1_1_0.md`:

```bash
demisto-sdk update-release-notes -i Packs/Cybersixgill-ActionableAlerts -u minor
```

3) Regarding the [unit test build failure](https://app.circleci.com/pipelines/github/demisto/content/146926/workflows/358384f7-bcde-4bc1-80ac-17d1938bead6/jobs/486007), this is an internal issue that was fixed in our SDK so no need to worry about it.

**Comment by tamarsix** [comment]
@kgal-pan Thanks, we still working on this PR - the status is in progress

**Comment by kgal-pan** [comment]
@tamarsix @Umamahesh-Loginsoft - Please fix the following:

1) Split the unified YAML`script-CybersixgillActionableAlertStatusUpdate.yml` using:

```bash
demisto-sdk split -i Packs/Cybersixgill-ActionableAlerts/Scripts/script-CybersixgillActionableAlertStatusUpdate.yml -o Packs/Cybersixgill-ActionableAlerts/Scripts/
```

2) Sync form with `master`:

```bash
./Utils/git_pull_master_into_fork.sh
```

3) Update Docker image in `CybersixgillActionableAlerts.yml:23` to `1.0.0.26756`.

4) Fix incident field `Cybersixgill_Status` JSON format:

```bash
demisto-sdk format -i Packs/Cybersixgill-ActionableAlerts/IncidentFields/Cybersixgill_Status.json
```

In addition, [please sign the CLA](https://cla-assistant.io/demisto/content?pullRequest=17825).

**Comment by tamarsix** [comment]
@kgal-pan regarding the docker image, is this the one built for https://github.com/demisto/dockerfiles/pull/7031? Cause this PR wasn’t merged yet

**Comment by kgal-pan** [comment]
@tamarsix - It's just the latest Docker image available in Dockerhub with the `sixgill` tag:

image

**Comment by Umamahesh-Loginsoft** [comment]
@kgal-pan I pushed the latest code after modifications could you please check it once and let me know if anything needs to changed

**Comment by kgal-pan** [comment]
> @kgal-pan I pushed the latest code after modifications could you please check it once and let me know if anything needs to changed

@Umamahesh-Loginsoft - Please sign the [Contribution Form](https://cla-assistant.io/demisto/content?pullRequest=17825)

**Comment by Umamahesh-Loginsoft** [comment]
> > @kgal-pan I pushed the latest code after modifications could you please check it once and let me know if anything needs to changed
> 
> @Umamahesh-Loginsoft - Please sign the [Contribution Form](https://cla-assistant.io/demisto/content?pullRequest=17825)

@kgal-pan signed in with Contribution form

**Comment by kgal-pan** [comment]
@Umamahesh-Loginsoft - Did you do this as well?

> 4) Fix incident field Cybersixgill_Status JSON format:
> ``` bash
> demisto-sdk format -i Packs/Cybersixgill-ActionableAlerts/IncidentFields/Cybersixgill_Status.json
> ```

I don't see this file was touched in the 2 latest commits.

**Comment by Umamahesh-Loginsoft** [comment]
> @Umamahesh-Loginsoft - Did you do this as well?
> 
> > 4. Fix incident field Cybersixgill_Status JSON format:
> > 
> > ```shell
> > demisto-sdk format -i Packs/Cybersixgill-ActionableAlerts/IncidentFields/Cybersixgill_Status.json
> > ```
> 
> I don't see this file was touched in the 2 latest commits.

@kgal-pan I pushed it now Sorry Previously I forgot to push this file

**Comment by bziser** [comment]
@Umamahesh-Loginsoft 
Hi, thank you for your contribution!
There is an issue with the build.
It can be fixed by entering https://app.circleci.com/pipelines/github/demisto/content?filter=mine) and disconnecting. You will then need to connect to your GitHub account again and push the code.

**Comment by Umamahesh-Loginsoft** [comment]
> @Umamahesh-Loginsoft Hi, thank you for your contribution! There is an issue with the build. It can be fixed by entering this [link](https://app.circleci.com/pipelines/github/demisto/content?filter=mine) and disconnecting. You will then need to connect to your GitHub account again and push the code.

@bziser I am getting 404 error I am click on link provided by you

**Comment by bziser** [comment]
> > @Umamahesh-Loginsoft Hi, thank you for your contribution! There is an issue with the build. It can be fixed by entering this [link](https://app.circleci.com/pipelines/github/demisto/content?filter=mine) and disconnecting. You will then need to connect to your GitHub account again and push the code.
> 
> @bziser I am getting 404 error(Page Not Found) If I am clicking on the link provided by you

Would you be able to check if there is some application or VPN blocking your access to this site? I can access it even when I'm not connected.

**Comment by Umamahesh-Loginsoft** [comment]
> > > @Umamahesh-Loginsoft Hi, thank you for your contribution! There is an issue with the build. It can be fixed by entering this [link](https://app.circleci.com/pipelines/github/demisto/content?filter=mine) and disconnecting. You will then need to connect to your GitHub account again and push the code.
> > 
> > 
> > @bziser I am getting 404 error(Page Not Found) If I am clicking on the link provided by you
> 
> @Umamahesh-Loginsoft Would you be able to check if there is some application or VPN blocking your access to this site? I can access it even when I'm not connected.

@bziser I am able to access it what should I do now?
image

**Comment by bziser** [comment]
> > > > @Umamahesh-Loginsoft Hi, thank you for your contribution! There is an issue with the build. It can be fixed by entering this [link](https://app.circleci.com/pipelines/github/demisto/content?filter=mine) and disconnecting. You will then need to connect to your GitHub account again and push the code.
> > > 
> > > 
> > > @bziser I am getting 404 error(Page Not Found) If I am clicking on the link provided by you
> > 
> > 
> > @Umamahesh-Loginsoft Would you be able to check if there is some application or VPN blocking your access to this site? I can access it even when I'm not connected.
> 
> @bziser I am able to access it what should I do now? image

@Umamahesh-Loginsoft 
As you can see, you need to disconnect by pressing the left upper button:
image

You will need to log in with your GitHub account again and git push again.

**Comment by kgal-pan** [comment]
@Umamahesh-Loginsoft - Please run the following commands:

```bash
rm Packs/Cybersixgill-ActionableAlerts/Scripts/script-CybersixgillActionableAlertStatusUpdate.yml

demisto-sdk format -i Packs/Cybersixgill-ActionableAlerts/Scripts/CybersixgillActionableAlertStatusUpdate/CybersixgillActionableAlertStatusUpdate.yml
```

In addition, please replace the script ID with the script name (`CybersixgillActionableAlertStatusUpdate`) in `Packs/Cybersixgill-ActionableAlerts/IncidentFields/Cybersixgill_Status.json:36`.

File: Packs/Cybersixgill-ActionableAlerts/Integrations/CybersixgillActionableAlerts/CybersixgillActionableAlerts.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def get_incident_init_params():
         'threat_level': demisto.params().get('threat_level', None),
         'threat_type': demisto.params().get('threat_type', None)
     }
-    return {param_k: param_v for param_k, param_v in params_dict.items() if param_v is not None}
+    return {param_k: param_v for param_k, param_v in params_dict.items() if param_v is not None and param_v}
```
Comment (bziser) [inline Packs/Cybersixgill-ActionableAlerts/Integrations/CybersixgillActionableAlerts/CybersixgillActionableAlerts.py:48]:
```suggestion
    return {param_k: param_v for param_k, param_v in params_dict.items() if param_v}
```
While not mandatory, it is more clear and does the same thing.

File: Packs/Cybersixgill-ActionableAlerts/Integrations/CybersixgillActionableAlerts/CybersixgillActionableAlerts.py [hunk 2]
```diff
@@ -167,45 +167,72 @@ def fetch_incidents():
...
-    demisto.info(f'Adding {len(incidents)} to demisto')
-    demisto.incidents(incidents)
-
-    if len(incidents):
-        demisto.info(f'Update last fetch time to: {newest_incident_date}')
-        demisto.setLastRun({
-            'last_fetch_time': newest_incident_date
-        })
+
+    if len(items) >= 0:
```
Comment (bziser) [inline Packs/Cybersixgill-ActionableAlerts/Integrations/CybersixgillActionableAlerts/CybersixgillActionableAlerts.py:171]:
```suggestion
    if len(items) > 0:
```
On the next line, you use `items[0]` which will cause "index out of range" if `len(items) == 0`

File: Packs/Cybersixgill-ActionableAlerts/Scripts/CybersixgillActionableAlertStatusUpdate/CybersixgillActionableAlertStatusUpdate.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+from urllib.parse import parse_qs, urlparse
```
Comment (bziser) [inline Packs/Cybersixgill-ActionableAlerts/Scripts/CybersixgillActionableAlertStatusUpdate/CybersixgillActionableAlertStatusUpdate.py:1]: Not mandatory:
This code will be easier to understand if it is inserted into the "main" function.

**Comment by bziser** [comment]
Looks good.
Thank you for your contribution!
Please check my comments and merge from master.

**Comment by bziser** [comment]
@tamarsix I need you to pull from origin/master.
Please run `./Utils/git_pull_master_into_fork.sh`.

**Comment by tamarsix** [comment]
@bziser Hey, both @Umamahesh-Loginsoft and I getting permission denied error when trying to run the script

**Review by bziser — APPROVED** [review]

**Pull request merged**